native/android/SwrveSDKPushSupport/src/amazon/java/com/swrve/unity/adm/SwrveAdmIntentService.java

```diff
@@ -0,0 +1,342 @@
+        String msgText = msg.getString("text");
+        if (!SwrveAdmHelper.isNullOrEmpty(msgText)) {
+            // Build notification
+            NotificationCompat.Builder mBuilder = createNotificationBuilder(msgText, msg);
+            mBuilder.setContentIntent(contentIntent);
+            return mBuilder.build();
+        }
+        return null;
+    }
+
+    private NotificationCompat.Builder createNotificationBuilder(String msgText, Bundle msg) {
+        Context context = getApplicationContext();
+        SharedPreferences prefs = SwrveAdmPushSupport.getAdmPreferences(context);
+        Resources res = getResources();
+        String pushTitle = prefs.getString(SwrveAdmPushSupport.PROPERTY_APP_TITLE, null);
+        String iconResourceName = prefs.getString(SwrveAdmPushSupport.PROPERTY_ICON_ID, null);
+        String materialIconName = prefs.getString(SwrveAdmPushSupport.PROPERTY_MATERIAL_ICON_ID, null);
+        String largeIconName = prefs.getString(SwrveAdmPushSupport.PROPERTY_LARGE_ICON_ID, null);
+        int accentColor = prefs.getInt(SwrveAdmPushSupport.PROPERTY_ACCENT_COLOR, -1);
+
+        PackageManager packageManager = context.getPackageManager();
+        ApplicationInfo app = null;
+        try {
+            app = packageManager.getApplicationInfo(getPackageName(), 0);
+        } catch (Exception exp) {
+            exp.printStackTrace();
+        }
+
+        int iconId = 0;
+        if (SwrveAdmHelper.isNullOrEmpty(iconResourceName)) {
+            // Default to the application icon
+            if (app != null) {
+                iconId = app.icon;
+            }
+        } else {
+            iconId = res.getIdentifier(iconResourceName, "drawable", getPackageName());
+        }
+
+        int finalIconId = iconId;
+        boolean mustUseMaterialDesignIcon = (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.LOLLIPOP);
+        if (SwrveAdmHelper.isNullOrEmpty(materialIconName)) {
+            // No material (Android L+) icon configured
+            Log.w(TAG, "No material icon specified. We recommend setting a special material icon for Android L+");
+        } else if(mustUseMaterialDesignIcon) {
+            // Running on Android L+
+            finalIconId = res.getIdentifier(materialIconName, "drawable", getPackageName());
+        }
+
+        if (SwrveAdmHelper.isNullOrEmpty(pushTitle)) {
+            if (app != null) {
+                // No configured push title
+                CharSequence appTitle = app.loadLabel(packageManager);
+                if (appTitle != null) {
+                    // Default to the application title
+                    pushTitle = appTitle.toString();
+                }
+            }
+            if (SwrveAdmHelper.isNullOrEmpty(pushTitle)) {
+                pushTitle = "Configure your app title";
+            }
+        }
+
+        // Build notification
+        NotificationCompat.Builder mBuilder = new NotificationCompat.Builder(this)
```

Comment:
Yeah... I wouldn't agree with it there, either!  "m" is old school C / C++ standard for member variable, and I didn't even like them back then.  This ain't a member variable, and this is Java.